relentless/core.py

```diff
@@ -40,6 +49,13 @@ class Interpolator:
 
     """
     def __init__(self, x, y):
+        x = np.atleast_1d(x)
+        if isinstance(x, (float,int)) or isinstance(y, (float,int)):
+            raise ValueError('x and y must not be scalars')
+        if isinstance(x[0], (tuple,list,np.ndarray)) or isinstance(y[0], (tuple,list,np.ndarray)):
+            raise ValueError('x and y must be 1-dimensional')
```

Comment:
Since you have already cast x to an array, just use the array shape to simplify these tests. (Usually enumerating types is a bad idea.) First, you should coerce both x and y into arrays. Then, you should test the dimension and shape of x, then check that x and y have the same shape. Like this:
```py
x = np.atleast_1d(x)
y = np.atleast_1d(y)
# 1d
if x.ndim > 1:
    raise
# non-scalar
if x.shape[0] == 1:
    raise
# x and y are same shape
if x.shape != y.shape:
    raise
```